transmart-core-api/src/main/groovy/org/transmartproject/core/multidimquery/MultiDimensionalDataResource.groovy

```diff
@@ -39,6 +42,15 @@ interface MultiDimensionalDataResource {
      */
     Iterable getDimensionElements(Dimension dimension, Constraint constraint, User user)
 
+    /**
+     * Parse a sort specification to an ordered map of dimensions to SortOrders
+     * @param sort the sort order as an ordered map or list of pairs, with keys and values as objects or strings
+     * @return The sort as an ordered map
+     *
+     * TODO: parseSort should live in rest-api, not here
+     */
+    Map<Dimension, SortOrder> parseSort(sort)
```

Comment:
What is the type of sort?

Reply:
`List<String | Dimension | List<String | Dimension | SortOrder> | Map<String | Dimension, String  | SortOrder>>`
But I don't think Groovy can represent that :). This parses the sort order from a json structure to the internal representation.